File: Number_Of_Steps_To_Reduce_A_Number_To_Zero.py
class Solution:
    def numberOfSteps (self, num: int) -> int:
        ans = 0
        while num != 0:
            
            ans += 1
            
            if num % 2 == 0:
                
                num //= 2
                
            else:
                
                num -= 1
                
        return ans

File: Number-Of-Steps-To-Reduce-A-Number-To-Zero.java
class Solution {
    public int numberOfSteps(int num) {
        int steps = 0;
        
        while (num > 0) {
            if (num % 2 == 1)
                num -= 1;
            else
                num /= 2;
            
            steps++;
        }
        
        return steps;
    }
}
